0035-spsp-invoices/0035-spsp-invoices.md

```diff
@@ -0,0 +1,124 @@
+
+### Operation Overview
+
+The Payer's SPSP Client will set up a [STREAM](../0029-stream/0029-stream.md) connection to the Payee's SPSP Server using the [Payment Pointer](../0026-payment-pointers/0026-payment-pointers.md) resolving to the Invoice SPSP Endpoint, as described by the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). On connection, the SPSP Client pushes as much value to the SPSP Server as necessary to complete the Invoice.
+
+## Model of Operation
+
+### Creating an Invoice SPSP Endpoint
+
+Prior to the Invoice Payment, the Payee has to create the Invoice SPSP Endpoint on the SPSP Server, which is represented by a [Payment Pointer](../0026-payment-pointers/0026-payment-pointers.md). This Payment Pointer MUST contain an ID representing the Invoice. The Payee's SPSP Server SHOULD store the Invoice details, which are defined in the `invoice`-object within the [Response Body](#Response-Body), within the Invoice SPSP Endpoint in order to account for balance changes.
+
+### Conducting an Invoice Payment
+
+The Payer's SPSP Client opens a [STREAM](../0029-stream/0029-stream.md) connection to the Payee's SPSP Server as described in the [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md). Once this connection is established, the process continues as follows: 
+
+The SPSP Client begins sending ILP packets to complete the Invoice.
+  1. The SPSP Client will adjust their `sendMax` to reflect the amount they're willing to send.
+      * `sendMax` SHOULD be derived from the Invoice.
+  2. The SPSP Server will adjust their `receiveMax` to reflect the amount they're willing to receive.
+      * `receiveMax` SHOULD be derived from the Invoice.
+  3. The SPSP Client's and Server's [STREAM Modules](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions) will move as much value as possible while staying inside these bounds.
+  4. If the SPSP Client reaches their `sendMax`, they end the stream and the connection. If the SPSP Server reaches their `receiveMax`, they will end the stream and the connection.
+
+The STREAM parameters - `sendMax` and `receiveMax` - are defined in [STREAM's frame encoding](../0029-stream/0029-stream.md#53-frames).
+
+
+## Specification
+
+### Query (`GET <SPSP Endpoint>`)
+
+The SPSP Client queries the Invoice SPSP Endpoint to get information about the SPSP Server as well as Invoice specific details:
+
+#### Request
+
+``` http
+GET /invoice123 HTTP/1.1
+Host: example.com
+Accept: application/spsp4+json
+```
+
+#### Response
+
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/spsp4+json
+
+{
+  "destination_account": "example.ilpdemo.red~invoice123",
+  "shared_secret": "6jR5iNIVRvqeasJeCty6C+YB5X9FhSOUPCL/5nha5Vs=",
+  "balance": "5360",
```

Comment:
Why isn't `balance` inside of the `invoice` object?

Reply:
It makes it consistent with the pull payment spec. Also, the variables within the `invoice` objects don't change on incoming money whereas balance does.